tools/lib/logreader.py

```diff
@@ -173,10 +170,11 @@ def _parse_identifiers(self, identifier: str | List[str]):
         return direct_source(identifier)
 
     sr = SegmentRange(parsed)
+    route = Route(sr.route_name)
     mode = self.default_mode if sr.selector is None else ReadMode(sr.selector)
     source = self.default_source if source is None else source
 
-    return source(sr, mode)
+    return source(sr, route, mode)
 
   def __init__(self, identifier: str | List[str], default_mode=ReadMode.RLOG, default_source=auto_source, sort_by_time=False, only_union_types=False):
```

Comment:
reset is run twice when using the log reader on one q/rlog one time, is this desired? It causes duplicate API queries

Reply:
reset in __iter__ isn't required anymore, fixed